src/modules/wallet/components/request/request.js

```diff
@@ -1,21 +1,50 @@
-import React from 'react';
+/* eslint-disable max-statements */
+import React, { useEffect, useMemo, useReducer, useRef } from 'react';
 import { regex } from 'src/const/regex';
 import { maxMessageLength } from '@transaction/configuration/transactions';
-import { mockAppTokens } from '@tests/fixtures/token';
+import { useApplicationExploreAndMetaData } from '@blockchainApplication/manage/hooks';
+import { useTransferableTokens } from '@token/fungible/hooks';
 import { validateAmountFormat } from 'src/utils/validators';
+import { getLogo } from 'src/redux/actions';
 import { sizeOfString } from 'src/utils/helpers';
 import { Input } from 'src/theme';
-import blockchainApplicationsExplore from '@tests/fixtures/blockchainApplicationsExplore';
 import Converter from 'src/modules/common/components/converter';
 import { useCurrentAccount } from 'src/modules/account/hooks';
 import i18n from 'src/utils/i18n/i18n';
 import Div from '@common/components/div/Div';
+import { useTranslation } from 'react-i18next';
 import MessageField from '../../../token/fungible/components/MessageField';
 import MenuSelect, { MenuItem } from '../MenuSelect';
 import RequestWrapper from './requestWrapper';
 import styles from './request.css';
 import WalletVisual from '../walletVisual';
-import chainLogo from '../../../../../setup/react/assets/images/LISK.png';
+
+const requestInitState = {
+  amount: {
+    error: false,
+    value: '',
+    loading: false,
+    feedback: '',
+  },
+  reference: {
+    error: false,
+    value: '',
+    loading: false,
+    feedback: '',
+  },
+  token: {
+    error: false,
+    value: '',
+    loading: false,
+    feedback: '',
+  },
+  recipientChain: {
+    error: false,
+    value: '',
```

Comment:
Would be nice if type of value wasnt mixed in the requestInitState. From the glance of it it looks like 
recipientChain.value represents a string holder but is in fact an object. So either 
`value: {},`
or
`value: undefined,`
would make more sense imo